RELEASING.md

```diff
@@ -0,0 +1,60 @@
+# Release Checklist
+
+These steps assume that you've checked out the Soketto repository and are in the root directory of it.
+
+1. Ensure that everything you'd like to see released is on the `develop` branch.
+
+2. Create a release branch off `develop`, for example `release-v0.6.0`. The branch name should start with `release`
+   in order that we can target commits with CI. You'll want to decide how far the version needs to be bumped 
+   based on the changes to date. If unsure what to bump the version to (eg is it a major, minor or patch 
+   release), check with the Parity Tools team.
+
+3. Check that you're happy with the current documentation.
+   
+   ```
+   cargo doc --open
+   ```
+
+   CI checks for broken internal links at the moment. Optionally you can also confirm that any external links
+   are still valid like so:
+
+   ```
+   cargo install cargo-deadlinks
+   cargo deadlinks --check-http
+   ```
+
+   If there are minor issues with the documentation, they can be fixed in the release branch.
+
+4. Bump the crate version in `Cargo.toml` to whatever was decided in step 2.
+
+5. Update `CHANGELOG.md` to reflect the difference between this release and last. If you're unsure of
+   what to add, check with the Tools team. 
+   
+   One way to gain some inspiration about what to write is by looking through the commit history since 
+   the last version (eg `git log --pretty LAST_VERSION_TAG..HEAD`).
+
+   Alternately, look at the commit history: https://github.com/paritytech/soketto/commits/develop.
+
+6. Commit any of the above changes to the release branch and open a PR in GitHub with a base of `develop`.
+
+7. Once the branch has been reviewed, we are ready to publish this branch to crates.io.
```

Comment:
```suggestion
7. Once the branch has been reviewed and merged to `develop`, we are ready to publish this branch to crates.io.
```

…or is it `master`?

Reply:
I guess I see the release branch as a snapshot in time that we're confident/happy with, and think that we should release and publish the exact commit that we "signed off" rather than do any merging first (which, to develop, would risk bringing in unintended changes). 

Merging to master first; I could go either way on that! I don't see a strong reason to do so if we tag release commits, but *shrug*, could go either way :)